packages/client/src/components/app/dynamic-filter/DynamicFilter.svelte

```diff
@@ -61,8 +61,9 @@
       clonedSchema = schema
     } else {
       allowedFields?.forEach(field => {
-        if (schema[field]) {
-          clonedSchema[field] = schema[field]
+        if (schema[field.name]) {
+          clonedSchema[field.name] = schema[field.name]
+          clonedSchema[field.name].displayName = field.displayName
```

Comment:
Same as above, this needs to account for the old structure of flat arrays of strings.